### DISCUSSION TITLE
Method names still visible after maximum obfuscation

### DISCUSSION THREAD

**Discussion by azzidoqsc** [mkaring/ConfuserEx#469] [Q&A]
Hello,

I am using ConfuserEx for my project and I really like it, I thank you to the team.
Today I have installed the latest release 1.6.0, I configured all the settings to the maximum, I confused the app then I imported the result in dnSpy for checking it. I observed that the names are not obfuscated. Do you know why ?
![image](https://user-images.githubusercontent.com/28946262/160293219-bf6ee0f6-728f-4856-a8ca-bd94ee562bf4.png)

![image](https://user-images.githubusercontent.com/28946262/160293237-b9a2d5ac-7c1e-4693-8640-bcec083757a7.png)

Thank you very much

**Comment by jkarpio** [comment]
Some public names cannot be obfuscated because it could made your binaries useless

**Reply by azzidoqsc** [reply]
Oh, yes, you are perfectly right.

**Reply by azzidoqsc** [reply]
One more question if it is allowed in this thread:
How secure is an application after confusing it with all protections set to aggressive ? I mean, are there tools that can reconstruct the code by just importing the confused app ? I usually use dnSpy to check the result, it looks confused and hard to reconstruct . But I have seen some videos of how some people did many steps in dnSpy and finally managed to reconstruct the code, but it was not straigforward.

**Reply by jkarpio** [reply]
Depending how big or important is your project but normally is to much work, in my opinion the obfuscation is pretty good

**Reply by mkaring** [reply]
With enough dedication and afford you can revert _some_ of the obfuscations. How ever not all of them. Things like `anti tamper`, `constants`, `ref proxy`, `resources` and the `type scamble` can be reverted if someone _really_ wants to spend the time.

`rename` is a one way road. There is no way to reconstruct the original names.

You'll find a description of the different protections, what they do and what security they offer in issue: #64

**Reply by azzidoqsc** [reply]
Thank you Martin. My application has all obfuscation levels configured and set to Aggressive in confuserEx. Sometimes the antivirus detects the application as infected but I explain my customers to add an exception.

**Reply by jkarpio** [reply]
Some virus and malicious software use confuser for protect how their work, because of this some AI in AV may detect your software as a virus because obfuscation is similar to already known viruses, my advice is send your software to the AV company as false positive for future updates in AV, also sing your binaries with a tool may help

**Reply by jkarpio** [reply]
> Thank you Martin. My application has all obfuscation levels configured and set to Aggressive in confuserEx. Sometimes the antivirus detects the application as infected but I explain my customers to add an exception.

Another option that worked before for me, was not obfuscate your .exe and do it for the rest of binaries but this depends on your app design, in my case exe file is very simple class for startup the business logic

**Reply by azzidoqsc** [reply]
Martin, is there any way to contact you in private ? I really need your help for a commercial C# app.
